funs/database_fun.R

```diff
@@ -67,46 +117,80 @@ download_project <- function(project, Tstart, Tend) {
         as_tibble()
 
     log_debug(paste("Got", nrow(stations), "stations for project", project))
-    apply(stations, 1, FUN  = insert_location_info_vectorized, conn = pool)
+    apply(stations, 1, FUN  = insert_location_info_vectorized, conn = conn)
 
 
-    v1 <- apply(sensors_meta, 1, FUN=store_sensor, conn=pool)
+    v1 <- apply(sensors_meta, 1, FUN=store_sensor, conn=conn)
 
 
-datastreams <- projinfo$datastream_data
+    datastreams <- projinfo$datastream_data
 
-for(i in unique(datastreams$kit_id)) {
-    kit <- datastreams %>%
-        filter(kit_id == i) %>%
-        select(-kit_id)
+    for(i in unique(datastreams$kit_id)) {
+        kit <- datastreams %>%
+            filter(kit_id == i) %>%
+            select(-kit_id)
 
 
-    type <- "datastream"
-    if(!doc_exists(type, ref = i, conn = pool)) {
-        log_trace("storing stream data info for {i}")
-        add_doc(type, ref = i, doc = kit, conn = pool)
-    } else {
-        log_trace("skipping store stream data info for {i}")
+        type <- "datastream"
+        if(!doc_exists(type, ref = i, conn = conn)) {
+            log_trace("storing stream data info for {i}")
+            add_doc(type, ref = i, doc = kit, conn = conn)
+        } else {
+            log_trace("skipping store stream data info for {i}")
 
-    }
+        }
 
-}
+    }
 
 
 }
 
-get_stations_from_project <- function(project) {
-    # this function gets all the stations belonging to a project. It
-    # returns a vector with stations ids (kit_ids), this vector can
+
+get_stations_from_selection <- function(name, type, conn = pool) {
+    # this function gets all the stations belonging to the selected
+    # project of municipality. 
+    # It returns a vector with stations ids (kit_ids), this vector can
     # then be used to download measurement data
+    # The type arguments determine if data is requested for a
+    # municpality or a project.
+    # arguments:
+    #    name: name of project or municipality
+    #    type: either 'project' or 'municipality'
+    #    conn: db connection object
+
+
+    switch(type, 
+           project = {
+               info <- get_doc(type = "project", ref = name, conn = conn)
+               kits <- info$sensor_data %>%
+                   pull(kit_id)
+           }, 
+           municipality = {
+               info <- get_doc(type = "municipality", ref = name, conn = conn)
+               kits <- info$sensor_data %>%
+                   pull(kit_id)
+           },
+           { #unknown type
+               stop("download_sensor_meta: unknown type")
+           })
 
-    info <- get_doc(type = "project", ref = project, conn = pool)
-    kits <- info$sensor_data %>%
-        pull(kit_id)
 
     return(kits)
 }
 
+# 
+# get_stations_from_project <- function(project) {
```

Comment:
Do we still need this code?

Reply:
No, removed in https://github.com/rivm-syso/Analyse-Together/pull/75/commits/9206d668e39d900653d740d69f79bce4de347999